lambdas/mail_send.js

```diff
@@ -4,17 +4,47 @@ if (process.env.NODE_ENV === 'development') {
 
 const domain = process.env.MAILGUN_DOMAIN;
 const apiKey = process.env.MAILGUN_API_KEY;
+const secret_key = process.env.SECRET_SEND_KEY;
 
+const { send } = require('micro');
+const statuses = require('statuses');
+const { upload } = require('micro-upload')
 const mailgun = require('mailgun-js');
 
-const mg = mailgun({ apiKey, domain });
-const data = {
-  from: 'WebConf <no-reply@webconf.tech>',
-  to: 'gahs94@gmail.com',
-  subject: 'WebConf • Nuestro newsletter 🙈',
-  text: 'Hola perrix! :D'
-};
+module.exports = upload(async (req, res) => {
+  if (!(req.files && req.files.file && req.files.file.mimetype === 'text/html')) {
+    return send(res, statuses['bad request'], 'Error - No file uploaded or wrong file format');
+  }
 
-module.exports = (req, res) => {
-  mg.messages().send(data, (error, body) => console.log(body));
-};
+  const { listName, subject, specificAddress, secret } = req.body;
+  if (secret === secret_key) {
+    const messageData = {
+      subject,
+      to: specificAddress,
+      html: req.files.file.data.toString('utf8'),
+      from: 'WebConf <no-reply@webconf.tech>'
+    };
+    if (listName) {
+      try {
+        const listMembers = await mailgun({ domain, apiKey }).lists(listName).members().list();
+        const subscribedMembers = listMembers.items.filter(user => user.subscribed);
+
+        messageData.to = subscribedMembers.map(user => user.address);
+        messageData['recipient-variables'] = subscribedMembers.reduce(
+          (accumulator, user) =>
+            ({ ...accumulator, [user.address]: user }), {}
+        );
+      } catch (error) {
+        return send(res, statuses['bad request'], 'Error - On list load ' + JSON.stringify(error));
```

Comment:
1. Template literals.
2. Do yo know if the error you receive is an object or an `Error` (I know, inheritance...you know what I mean)? Because if it's an object I would recommend you to do something like `{...error, _someCustomProp: 'On list load'}`; but if it's an `Error`, you should do `{ _someCustomProp: `On list load: ${error.message}}`. The point is, be consistent with the response type. (And yes, everything _stringified_).

Reply:
No underscore-prefixed properties, please.

Also, we're returning a failure response from an API, so matching against the Error type has no relevance here. API clients don't necessarily will create an Error object from these properties.

Reply:
> No underscore-prefixed properties, please.

Always, it was just to explain that it should be something custom that's not on the standard response.

> ...so matching against the Error type has no relevance here

This error comes from the Mailgun library, that's why I asked if he knew if it was an object or an Error; the idea is to get the proper message to return.

> **API clients don't necessarily** will create an Error object from these properties.

Come on, you never know how a library decide to handle this kind of stuff.

Also, [this is an `Error`](https://github.com/bojand/mailgun-js/blob/master/lib/request.js#L327).

Reply:
@homer0 

> Come on, you never know how a library decide to handle this kind of stuff.

Yes, hopefully from our frontend client, we'll know how to handle it :P